modules/products/idea/src/main/kotlin/com/metalbear/mirrord/products/idea/IdeaRunConfigurationExtension.kt

```diff
@@ -1,19 +1,34 @@
 package com.metalbear.mirrord.products.idea
 
-import com.intellij.execution.Executor
 import com.intellij.execution.RunConfigurationExtension
 import com.intellij.execution.configurations.JavaParameters
 import com.intellij.execution.configurations.RunConfigurationBase
 import com.intellij.execution.configurations.RunnerSettings
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.process.ProcessListener
 import com.intellij.execution.target.createEnvironmentRequest
 import com.intellij.execution.wsl.target.WslTargetEnvironmentRequest
 import com.intellij.openapi.components.service
 import com.intellij.openapi.externalSystem.service.execution.ExternalSystemRunConfiguration
+import com.intellij.openapi.project.Project
 import com.metalbear.mirrord.*
+import java.util.concurrent.ConcurrentHashMap
 
 class IdeaRunConfigurationExtension : RunConfigurationExtension() {
+    /**
+     * mirrord env set in ExternalRunConfigurations. Used for cleanup the configuration after the execution has ended.
+     */
+    private val runningProcessEnvs = ConcurrentHashMap<Project, Set<String>>()
+
     override fun isApplicableFor(configuration: RunConfigurationBase<*>): Boolean {
-        return true
+        val applicable = !configuration.name.startsWith("Build ")
```

Comment:
Wonder if we might have false positives here? I'd at least document what this specifically tries to catch

Reply:
No idea, to be honest. This check was already there, in the other method :man_shrugging: